tests/pytests/integration/modules/test_pillar.py

```diff
@@ -366,3 +368,90 @@ def test_pillar_refresh_pillar_ping(salt_cli, salt_minion, key_pillar):
         val = ret.json
         assert key in val
         assert val[key] is True
+
+
+@slowTest
+def test_pillar_refresh_pillar_scheduler(salt_cli, salt_minion):
+    """
+    Ensure schedule jobs in pillar are only updated when values change.
+    """
+
+    top_sls = """
+        base:
+          '{}':
+            - test_schedule
+        """.format(
+        salt_minion.id
+    )
+
+    test_schedule_sls = """
+        schedule:
+          first_test_ping:
+            function: test.ping
+            run_on_start: True
+            seconds: 3600
+        """
+
+    test_schedule_sls2 = """
+        schedule:
+          first_test_ping:
+            function: test.ping
+            run_on_start: True
+            seconds: 7200
+        """
+
+    with temp_pillar_file("top.sls", top_sls):
```

Comment:
Under PyTest, you don't need to import and instead do `with pytest.helpers.temp_pillar_file`